op-challenger/game/fault/contracts/oracle.go

```diff
@@ -37,3 +42,35 @@ func (c PreimageOracleContract) AddGlobalDataTx(data *types.PreimageOracleData)
 	call := c.contract.Call(methodLoadKeccak256PreimagePart, new(big.Int).SetUint64(uint64(data.OracleOffset)), data.GetPreimageWithoutSize())
 	return call.ToTxCandidate()
 }
+
+func (c PreimageOracleContract) InitLPP(ctx context.Context, uuid *big.Int, partOffset uint32, claimedSize uint32) (txmgr.TxCandidate, error) {
+	call := c.contract.Call(methodInitLPP, uuid, partOffset, claimedSize)
+	return call.ToTxCandidate()
+}
+
+func (c PreimageOracleContract) AddLeavesLPP(ctx context.Context, uuid *big.Int, input []byte, stateCommitments [][32]byte, finalize bool) (txmgr.TxCandidate, error) {
```

Comment:
It feels like this API is probably too low level. I put some suggested APIs in https://github.com/ethereum-optimism/client-pod/issues/470 in this case:
```
AddLeaves(uuid *big.Int, leaves []Leaf, finalize bool)
```
The advantage of writing our own bindings is that we can do some translation so we expose a nice go API instead of just a direct mapping from the contract. Even if we don't want to pass a Leaf struct that combines the input and state commitments, the commitments should at least be `common.Hash` since that's the actual go type. I guess it depends a bit how we wind up using the `KeccakState` as to whether we wind up with a set of leaf input bytes or a single byte array already created.  Whatever fits best for our business logic should be what we pass in here and can then convert to what the contract wants internally.